iceoryx2/src/port/details/incoming_connections.rs

```diff
@@ -0,0 +1,321 @@
+    ) -> Result<Option<(ChunkDetails<Service>, Chunk)>, ReceiveError> {
+        let msg = "Unable to receive another sample";
+        match connection.receiver.receive() {
+            Ok(data) => match data {
+                None => Ok(None),
+                Some(offset) => {
+                    let details = ChunkDetails {
+                        connection: connection.clone(),
+                        offset,
+                        origin: connection.sender_port_id,
+                    };
+
+                    let offset = match connection
+                        .data_segment
+                        .register_and_translate_offset(offset)
+                    {
+                        Ok(offset) => offset,
+                        Err(e) => {
+                            fail!(from self, with ReceiveError::ConnectionFailure(ConnectionFailure::UnableToMapSendersDataSegment(e)),
+                                "Unable to register and translate offset from publisher {:?} since the received offset {:?} could not be registered and translated.",
+                                connection.sender_port_id, offset);
+                        }
+                    };
+
+                    Ok(Some((
+                        details,
+                        Chunk::new(&self.message_type_details, offset),
+                    )))
+                }
+            },
+            Err(ZeroCopyReceiveError::ReceiveWouldExceedMaxBorrowValue) => {
+                fail!(from self, with ReceiveError::ExceedsMaxBorrowedSamples,
+                    "{} since it would exceed the maximum {} of borrowed samples.",
+                    msg, connection.receiver.max_borrowed_samples());
+            }
+        }
+    }
+
+    pub(crate) fn receive(&self) -> Result<Option<(ChunkDetails<Service>, Chunk)>, ReceiveError> {
+        let to_be_removed_connections = unsafe { &mut *self.to_be_removed_connections.get() };
+
+        if let Some(connection) = to_be_removed_connections.peek() {
+            if let Some((details, absolute_address)) = self.receive_from_connection(connection)? {
+                return Ok(Some((details, absolute_address)));
+            } else {
+                to_be_removed_connections.pop();
+            }
+        }
+
+        for id in 0..self.len() {
+            if let Some(ref mut connection) = &mut self.get_mut(id) {
+                if let Some((details, absolute_address)) =
+                    self.receive_from_connection(connection)?
+                {
+                    return Ok(Some((details, absolute_address)));
+                }
+            }
+        }
+
+        Ok(None)
+    }
+
+    pub(crate) fn start_update_connection_cycle(&self) {
+        self.visitor.next_cycle();
+    }
+
+    pub(crate) fn update_connection(
+        &self,
+        index: usize,
+        sender_details: SenderDetails,
+    ) -> Result<(), ConnectionFailure> {
+        let create_connection = match self.get(index) {
```

Comment:
Just as food for thought, since it slowed my review a bit down to fully understand this.
Would it make sense to name the variable according the information it holds instead of how it will be used? In this case it would be `is_connected`/`has_connection`/`has_valid_connection`

Reply:
I renamed the variable in the follow up branch.